In [1]:
import pandas as pd

url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"

df = pd.read_csv(url)

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


## 1. Asimetria (Skewness)

La asimetria sirve para saber hacia que lado esta mas concentrada una distribucion. Si el resultado es positivo significa que la cola de la distribucion se extiende hacia la derecha, y si es negativo se extiende hacia la izquierda.

In [4]:
age_skewness = df["Age"].skew()

print("Asimetria de Age:", age_skewness)

Asimetria de Age: 0.38910778230082704


### Interpretacion

El resultado de la asimetria de Age es positivo, aproximadamente 0.39. Esto quiere decir que la distribucion tiene una ligera inclinacion hacia la derecha. La mayoria de los pasajeros se encuentran en edades jovenes y adultas, mientras que hay menos pasajeros de edades mas grandes. Por eso la cola de la distribucion se extiende hacia las edades mayores.

## 2. Curtosis (Kurtosis)

La curtosis sirve para ver que tan pesadas son las colas de una distribucion y si existen valores extremos. Un resultado positivo puede indicar que hay mas valores extremos que en una distribucion normal.

In [5]:
fare_kurtosis = df["Fare"].kurt()

print("Curtosis de Fare:", fare_kurtosis)

Curtosis de Fare: 33.39814088089868


### Interpretacion

El resultado de la curtosis de Fare es positivo y es aproximadamente 34. Esto indica que la distribucion tiene colas pesadas y que existen algunos valores muy alejados del resto. Esto si confirma la existencia de outliers que se pueden encontrar en las tarifas, ya que algunos pasajeros pagaron mucho mas que la mayoria.

## 3. Tabla Cruzada (Cross-Tabulation)

La funcion pd.crosstab() sirve para comparar dos variables y mostrar sus resultados en forma de tabla. En este caso voy a comparar la clase del pasajero con si sobrevivio o no.

In [6]:
tabla_cruzada = pd.crosstab(
    df["Pclass"],
    df["Survived"],
    rownames=["Clase del pasajero"],
    colnames=["Sobrevivio"]
)

tabla_cruzada

Sobrevivio,0,1
Clase del pasajero,,
1,80,136
2,97,87
3,372,119


### Interpretacion

Al cruzar Pclass con Survived se puede ver rapidamente que la clase del pasajero tuvo relacion con la supervivencia. En primera clase sobrevivieron 136 pasajeros y 80 no sobrevivieron. En segunda clase sobrevivieron 87 y 97 no sobrevivieron. En tercera clase solo sobrevivieron 119 pasajeros, mientras que 372 no sobrevivieron.

El patron que se puede observar es que los pasajeros de primera clase tuvieron mejores resultados de supervivencia, mientras que en tercera clase hubo una cantidad mucho mayor de pasajeros que no sobrevivieron.

## Conclusion

Con estas tres tecnicas pude resumir diferentes partes del dataset del Titanic. La asimetria de Age mostro que la distribucion de edades esta ligeramente inclinada hacia las edades mayores. La curtosis de Fare mostro que existen valores extremos en las tarifas. Por ultimo, la tabla cruzada permitio ver rapidamente que la clase del pasajero estaba relacionada con la supervivencia, especialmente al comparar primera y tercera clase.